docs/migrations/migrations_guide.md

```diff
@@ -0,0 +1,192 @@
+# 📚 Database Migrations Guide – Nixopus
+
+This guide explains how database migrations work in Nixopus, how to create and apply them, and best practices for using them effectively.
+
+---
+
+## 🧰 Tool We Use: `golang-migrate`
+
+Nixopus uses [`golang-migrate`](https://github.com/golang-migrate/migrate) to manage SQL-based database migrations in Go. This tool supports programmatic and CLI-based migration execution, and we’ve embedded it into our application lifecycle so that migrations run automatically on startup.
```

Comment:
We are using Bun ORM, so all of the migration management is also done via `Bun ORM` only.
Ref: https://bun.uptrace.dev/

Reply:
You can check the code for migration in `api/internal/storage/migration.go`